pilot/pkg/config/kube/clustertrustbundle/controller.go

```diff
@@ -0,0 +1,142 @@
+	stopChan            chan struct{}
+}
+
+// NewClusterTrustBundleController creates a new ClusterTrustBundleController
+func NewClusterTrustBundleController(kubeClient kube.Client, caBundleWatcher *keycertbundle.Watcher) *ClusterTrustBundleController {
+	c := &ClusterTrustBundleController{
+		client:          kubeClient,
+		stopChan:        make(chan struct{}),
+		caBundleWatcher: caBundleWatcher,
+	}
+
+	c.queue = controllers.NewQueue("clustertrustbundle controller",
+		controllers.WithReconciler(c.reconcileClusterTrustBundle),
+		controllers.WithMaxAttempts(maxRetries))
+
+	c.clustertrustbundles = kclient.NewFiltered[*certificatesv1alpha1.ClusterTrustBundle](kubeClient, kclient.Filter{
+		FieldSelector: "metadata.name=" + istioClusterTrustBundleName,
+		ObjectFilter:  kubeClient.ObjectFilter(),
+	})
+
+	c.clustertrustbundles.AddEventHandler(controllers.FilteredObjectSpecHandler(c.queue.AddObject, func(o controllers.Object) bool { return true }))
+
+	return c
+}
+
+// startCaBundleWatcher listens for updates to the CA bundle and queues a reconciliation of the ClusterTrustBundle
+func (c *ClusterTrustBundleController) startCaBundleWatcher(stop <-chan struct{}) {
+	id, watchCh := c.caBundleWatcher.AddWatcher()
+	defer c.caBundleWatcher.RemoveWatcher(id)
+	for {
+		select {
+		case <-watchCh:
+			c.queue.AddObject(&certificatesv1alpha1.ClusterTrustBundle{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: istioClusterTrustBundleName,
+				},
+			})
+		case <-stop:
+			return
+		}
+	}
+}
+
+func (c *ClusterTrustBundleController) reconcileClusterTrustBundle(o types.NamespacedName) error {
+	if o.Name == istioClusterTrustBundleName {
+		return c.updateClusterTrustBundle(c.caBundleWatcher.GetCABundle())
+	}
+	return nil
+}
+
+// Run starts the controller
+func (c *ClusterTrustBundleController) Run(stopCh <-chan struct{}) {
+	if !kube.WaitForCacheSync("clustertrustbundle controller", stopCh, c.clustertrustbundles.HasSynced) {
+		return
+	}
+	go c.startCaBundleWatcher(stopCh)
+	defer close(c.stopChan)
+
+	// queue an initial event
+	c.queue.AddObject(&certificatesv1alpha1.ClusterTrustBundle{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: istioClusterTrustBundleName,
+		},
+	})
+
+	c.queue.Run(stopCh)
+	controllers.ShutdownAll(c.clustertrustbundles)
+}
+
+// updateClusterTrustBundle updates the root certificate in the ClusterTrustBundle
+func (c *ClusterTrustBundleController) updateClusterTrustBundle(rootCert []byte) error {
+	bundle := &certificatesv1alpha1.ClusterTrustBundle{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: istioClusterTrustBundleName,
+		},
+		Spec: certificatesv1alpha1.ClusterTrustBundleSpec{
+			TrustBundle: string(rootCert),
+		},
+	}
+
+	existing, err := c.client.Kube().CertificatesV1alpha1().ClusterTrustBundles().Get(context.Background(), istioClusterTrustBundleName, metav1.GetOptions{})
+	if err == nil {
+		// Update existing bundle
+		existing.Spec.TrustBundle = string(rootCert)
```

Comment:
very nit: we could skip if its the same. For example, a new istiod on startup

Reply:
comparing and skipping if equal now